src/item.cpp

```diff
@@ -2432,11 +2432,17 @@ int item::volume( bool integral ) const
         }
     }
 
-// tool mods also add volume
-    if (has_flag("ATOMIC_AMMO")) {
+    // Battery mods also add volume
+    if( has_flag("ATOMIC_AMMO") ) {
         ret += 1;
     }
 
+    if( has_flag("DOUBLE_AMMO") ) {
+        // Batteries have volume 1 per 100 charges
+        // TODO: De-hardcode this
```

Comment:
```
/** Specific ammo data, returns nullptr if item is neither ammo nor loaded with any */
const itype * ammo_data() const;
```

`nullptr` for unloaded items.
I'd need to add a special case just for unloaded items and that could result in discrepancies where a loaded item could lose volume.